src/main/groovy/com/causecode/fileuploader/util/renewer/DefaultUFileTemporaryUrlRenewer.groovy

```diff
@@ -0,0 +1,120 @@
+    private final CDNFileUploader cdnFileUploader
+    private final int maxResultsInOneIteration
+    private final boolean forceAll
+
+    DefaultUFileTemporaryUrlRenewer(
+            CDNProvider cdnProvider,
+            CDNFileUploader cdnFileUploader,
+            boolean forceAll,
+            int maxResultsInOneIteration = 100) {
+
+        checkArgument(cdnProvider != null, 'CDNProvider can not be null')
+        checkArgument(cdnFileUploader != null, 'CDNFileUploader can not be null')
+        checkArgument(maxResultsInOneIteration > 0, 'maxResultsInOneIteration can\'t be negative or zero')
+
+        this.cdnProvider = cdnProvider
+        this.cdnFileUploader = cdnFileUploader
+        this.forceAll = forceAll
+        this.maxResultsInOneIteration = Math.min(maxResultsInOneIteration, 1000)
+    }
+
+    @Override
+    void renew() {
+        int offset = 0
+        List<UFile> uFiles
+
+        while ((uFiles = getResultListFromOffset(offset)).size()) {
+            processResultList(uFiles)
+            offset += this.maxResultsInOneIteration
+        }
+
+        cdnFileUploader.close()
+    }
+
+    private List<UFile> getResultListFromOffset(int offset) {
+        return UFile.createCriteria().list([offset: offset, max: this.maxResultsInOneIteration], criteriaClosure())
+    }
+
+    private Closure criteriaClosure() {
+        return {
+            eq('type', UFileType.CDN_PUBLIC)
+            eq('provider', this.cdnProvider)
+
+            if (Holders.flatConfig['fileuploader.persistence.provider'] == 'mongodb') {
+                eq('expiresOn', [$exists: true])
+            } else {
+                isNotNull('expiresOn')
+            }
+
+            if (!forceAll) {
+                or {
+                    lt('expiresOn', new Date())
+                    // Getting all CDN UFiles which are about to expire within one day.
+                    between('expiresOn', new Date(), new Date() + 1)
+                }
+            }
+        }
+    }
+
+    private List<UFile> processResultList(List<UFile> resultList) {
+        resultList.each { UFile uFile ->
+            log.debug "Renewing URL for $uFile"
+
+            try {
+                updateExpirationPeriodAndUrl(uFile)
+            } catch (StorageException ex) {
+                log.error("URL is not generated for File: $uFile due to $ex.message", ex)
+            }
+        }
+    }
+
+    @SuppressWarnings('CannotModifyReference') // Method just saves the given instance after updating the properties
+    private void updateExpirationPeriodAndUrl(UFile uFile) {
+        long expirationPeriod = getExpirationPeriod(uFile.fileGroup)
+
+        uFile.path = cdnFileUploader.getTemporaryURL(uFile.container, uFile.fullName, expirationPeriod)
+        uFile.expiresOn = new Date(new Date().time + expirationPeriod * 1000)
+
+        if (NucleusUtils.save(uFile, true)) {
+            log.debug "New URL for $uFile [$uFile.path] [$uFile.expiresOn]"
+        }
```

Comment:
May be you can throw an exception here if save call returns false. Catch and log it in line number 98.

Reply:
Changed method to return boolean.